### ISSUE TITLE
Bold Font Customisation

### ISSUE THREAD

**Issue by HadyMash** [microsoft/vscode#120425]
Hello, I hope you are doing well.

I would like to be able to better customise fonts in VS Code. Many fonts come with multiple font weights such as light, regular, medium, semi-bold, bold, etc. with some even having variable weights. I recently started using Fira Code instead of the default font and I am loving it. I am currently using the One Dark Pro theme and I customised the theme to have parameter variables to be a different colour an to be bold using `"fontStyle"`. The problem is that I find Fira Code's bold to be too bold and I think the medium weight is a much better alternative. The `"fontStyle"` parameter doesn't allow for any in between and it's either normal or bold. something like the `"editor.fontWeight"` option takes in normal, bold, but also takes values between 1 and 1000 to have an 'in between'.

I would like to have an option to set which font in the font family is the 'bold' so for example I can set what VS Code considers to be bold as the medium weight. An alternative solution would be to allow the textMateRules' settings' "fontStyle" to be similar to the editor.fontWeight where I can put in numbers and not just "bold".
Sorry if this is already a feature I am not aware of but I tried looking through VS Code's settings and I couldn't find anything. I also had a look online and couldn't find what I was looking for.

**Comment by alexdima** [comment]
@HadyMash Have you tried using `Fira Code Medium` as the font family? This works for me on macOS:


https://user-images.githubusercontent.com/5047891/114421074-370e4500-9bb5-11eb-8b8e-895ece3d24ba.mp4

I got the name from the Font Book on macOS:

![image](https://user-images.githubusercontent.com/5047891/114421128-47262480-9bb5-11eb-9f75-03a957c36adf.png)

**Comment by HadyMash** [comment]
Hello @alexdima
I don't want to use the Medium font weight throughout all of VS Code, I used the theme customisation options and to make parameter variables bold, the problem is that they are too bold. I want to be able to customise that.

This is what I did:
```
"editor.tokenColorCustomizations": {
	"[One Dark Pro]": {
		"textMateRules": [
			{
				"scope": "variable.parameter",
				"settings": {
					"foreground": "#c2c2c2",
					"fontStyle": "bold"
				},
			}
		]
	}
}
```

The problem is that the `"fontStyle"` doesn't allow for customisation of the 'boldness' of the font, that's what I would like to be able to customise, similar to the `"editor.fontWeight"` option.
Thanks for your help.
